components/cockpit.jsx

```diff
@@ -0,0 +1,173 @@
+import styled, {css} from "styled-components";
+import { useState } from "react";
+import { nanoid } from 'nanoid';
+
+export function Cockpit(){
+
+const [commandLine, setCommandLine] = useState([])
+const [count, setCount]= useState(0)
+
+function addRight(){
+if(count < 12){commandLine.push("RIGHT")
+setCommandLine(commandLine)
+setCount(commandLine.length)}}
+
+function addLeft(){
+if(count < 12){commandLine.push("LEFT")
+setCommandLine(commandLine)
+setCount(commandLine.length)}}
+
+function addUp(){
+if (count < 12){commandLine.push("UP")
+setCommandLine(commandLine)
+setCount(commandLine.length)}}
+
+function addDown(){
+if (count < 12){
+commandLine.push("DOWN")
+setCommandLine(commandLine)
+setCount(commandLine.length)}}
+
+function del(){
+commandLine.pop()
+setCommandLine(commandLine)
+setCount(commandLine.length)}
+
+function go(){
+console.log("go")}
+
+return(
+    <>
+<CockpitFrame>
+<Keyboard>
+<Key onClick={addUp}>UP</Key>
+<Key onClick={addDown}>DOWN</Key>
+<Key onClick={addLeft}>LEFT</Key>
+<Key onClick={addRight}>RIGHT</Key>
+<DelKey onClick={del}>DEL</DelKey>
+</Keyboard>
+<CommandLine>
+{commandLine.map((element)=>(<Command key={nanoid()}>{element}</Command>))}
```

Comment:
Calling `nanoid` directly within render to create keys is not a good idea. They will be different on every render which kinda defeats the point. Here even though it's rarely the case it's better to use the array index, since we don't have anything else (like an id) to uniquely identify each item.
 
```suggestion
{commandLine.map((element, index)=>(<Command key={index}>{element}</Command>))}
```

Reply:
alright, good to know as well!